src/core/server/http/http_server.ts

```diff
@@ -127,4 +137,31 @@ export class HttpServer {
     const routePathStartIndex = routerPath.endsWith('/') && routePath.startsWith('/') ? 1 : 0;
     return `${routerPath}${routePath.slice(routePathStartIndex)}`;
   }
+
+  private registerOnRequest = (fn: OnRequest) => {
```

Comment:
not directly related to Security integration directly, but interesting to check consistency in `registerOnRequest ` and `registerAuth` API

Reply:
I'm writing integration tests for this functionality, but send them as a separate PR  as I would like to have this PR merged

Reply:
Any reason this is defined as an arrow function instead of a class method?

Reply:
this function is [exposed](https://github.com/restrry/kibana/blob/c043c297ce6634881a886b0b685da5fdbe747ef0/src/core/server/http/http_server.ts#L89) as a part of `setup` contract. Otherwise, we either have to bind it in `setup` method or expose an unbound version.